src/vendor/snippet/snippetParser.ts

```diff
@@ -295,7 +295,7 @@ export class Choice extends Marker {
 
 	toTextmateString(): string {
 		return this.options
-			.map(option => option.value.replace(/\||,/g, '\\$&'))
+			.map(option => option.value.replace(/\||,|\\/g, '\\$&'))
```

Comment:
Codeql actually found a bug!  It is in the code we vendored in from VSCode; filed https://github.com/microsoft/vscode/pull/169287 to fix it upstream as well